client/src/components/QuestionBox.tsx

```diff
@@ -0,0 +1,94 @@
+import { useState } from "react";
+import { useQueryClient } from "@tanstack/react-query";
+import AnswersAndVotes from "./AnswersAndVoters";
+import Button from "./Button";
+import * as adminQuickRound from "../newtork/adminQuickRound";
+import EditQuestionForm from "./EditQuestionForm";
+import { Modal, ModalHeader, ModalBody } from "./Modal";
+
+interface Answer {
+  [key: string]: number;
+}
+
+interface QuestionProps {
+  id: number;
+  question: string;
+  answers: Answer;
+  createdAt: string;
+  updatedAt: string;
+}
+
+function convertDate(date: string) {
+  return new Date(date).toLocaleString();
+}
+
+export default function QuestionBox({ question }: { question: QuestionProps }) {
+  const queryClient = useQueryClient();
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  async function handleDelete(id: number) {
+    try {
+      await adminQuickRound.deleteQuestion(id);
+      queryClient.invalidateQueries({ queryKey: ["questions"] });
+    } catch (error) {
+      console.error(error);
+    }
+  }
+
+  function handleEdit() {
+    setIsModalOpen(true);
+  }
+
+  interface UpdatedDataProps {
+    question: string;
+    answers: { [key: string]: number };
+  }
+
+  const handleEditSubmit = async (data: UpdatedDataProps) => {
+    try {
+      await adminQuickRound.updateQuestion(question.id, data);
+      queryClient.invalidateQueries({ queryKey: ["questions"] });
+      setIsModalOpen(false);
+    } catch (error) {
+      console.error("Failed to update question", error);
+    }
+  };
+
+  return (
+    <>
+      <div className="border border-gray-400 p-8 rounded-lg shadow-[0_0_10px_rgba(0,0,0,0.3)] bg-indigo-300 w-full grid grid-cols-2 gap-20">
```

Comment:
If we end up with 100 or more questions, we’ll need to change the endpoint logic because loading all 100 at once would take too long. To fix this, we should add pagination, so we can show a smaller number of items per page.